packages/react-components/react-avatar/src/components/AvatarGroupOverflow/renderAvatarGroupOverflow.tsx

```diff
@@ -8,6 +11,20 @@ import type { AvatarGroupOverflowState, AvatarGroupOverflowSlots } from './Avata
 export const renderAvatarGroupOverflow_unstable = (state: AvatarGroupOverflowState) => {
   const { slots, slotProps } = getSlots<AvatarGroupOverflowSlots>(state);
 
-  // TODO Add additional slots in the appropriate place
-  return <slots.root {...slotProps.root} />;
+  return (
+    <slots.root {...(slotProps.root as PopoverProps)}>
+      <PopoverTrigger>
+        <Tooltip content={state.tooltipContent} relationship="label">
+          <slots.overflowButton {...slotProps.overflowButton} />
+        </Tooltip>
+      </PopoverTrigger>
+      {slots.overflowSurface && (
+        <slots.overflowSurface {...slotProps.overflowSurface}>
+          <AvatarGroupContext.Provider value={{ isOverflow: true, size: 24 }}>
+            {slots.overflowContent && <slots.overflowContent {...slotProps.overflowContent} />}
+          </AvatarGroupContext.Provider>
+        </slots.overflowSurface>
+      )}
```

Comment:
Is it possible to get rid of the null checks for these slots? They should never be null. 
```suggestion
        <slots.overflowSurface {...slotProps.overflowSurface}>
          <AvatarGroupContext.Provider value={{ isOverflow: true, size: 24 }}>
            <slots.overflowContent {...slotProps.overflowContent} />
          </AvatarGroupContext.Provider>
        </slots.overflowSurface>
```
Or, if the typings require a null check, then (nit) group the checks for `slots.overflowSurface && slots.overflowContent` together.